src/main/java/com/user_service/controllers/UserAuthenticationController.java

```diff
@@ -36,7 +36,47 @@ public ResponseEntity<String> confirmSignUp(@RequestBody ConfirmRegisterDto conf
     }
 
     @PostMapping(value = "/login")
-    public void signInUser(@RequestBody LoginUserDto loginUserDto) {
-        userAuthenticationService.signInUser(loginUserDto);
+    public ResponseEntity<String> signInUser(@RequestBody SignInUserDto signInUserDto) {
+        SignInResponseDto signInResponseDto = userAuthenticationService.signInUser(signInUserDto);
+
+        if (signInResponseDto.getAdminInitiateAuthResponse() == null) {
+            return new ResponseEntity<>(signInResponseDto.getMesssage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return new ResponseEntity<>("Login successful", HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/changePassword")
+    public ResponseEntity<String> changePassword(@RequestBody ChangePasswordDto changePasswordDto) {
+        ChangePasswordResponseDto changePasswordResponseDto = userAuthenticationService.changePassword(changePasswordDto);
+
+        if (changePasswordResponseDto.getChangePasswordResponse() == null) {
+            return new ResponseEntity<>(changePasswordResponseDto.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return new ResponseEntity<>("Password changed", HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/forgotPassword")
+    public ResponseEntity<String> forgotPassword(@RequestBody ForgotPasswordDto forgotPasswordDto) {
+        ForgotPasswordResponseDto forgotPasswordResponseDto = userAuthenticationService.forgotPassword(forgotPasswordDto);
+
+        if (forgotPasswordResponseDto.getForgotPasswordResponse() == null) {
+            return new ResponseEntity<>(forgotPasswordResponseDto.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return new ResponseEntity<>("Forgot password validation code send on email", HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/confirmForgotPassword")
+    public ResponseEntity<String> confirmForgotPassword(@RequestBody ConfirmForgotPasswordDto confirmForgotPasswordDto) {
+        ConfirmForgotPasswordResponseDto confirmForgotPasswordResponseDto = userAuthenticationService.confirmForgotPassword(confirmForgotPasswordDto);
+
+        if (confirmForgotPasswordResponseDto.getConfirmForgotPasswordResponse() == null) {
+            return new ResponseEntity<>(confirmForgotPasswordResponseDto.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return new ResponseEntity<>("Password reseted", HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/authorize")
+    public void authorize(@RequestBody AuthorizeDto authorizeDto) {
```

Comment:
I think that this belongs to be in a separate class. The other methods seem to be related to user sign-up/sing-in. This method on the other hand will be used internally, for service-to-service communication.